src/main/drivers/compass/compass_spi_hmc5883l.c

```diff
@@ -21,17 +21,22 @@
 
 #include <platform.h>
 
+#include "config/parameter_group.h"
+
 #include "drivers/io.h"
 #include "drivers/bus_spi.h"
 
 #include "compass.h"
 #include "compass_hmc5883l.h"
 
+#define SPI_CFG_TO_DEV(x) ((x) - 1)
+
 #ifdef USE_MAG_SPI_HMC5883
 
 #define DISABLE_HMC5883      IOHi(hmc5883CsPin)
 #define ENABLE_HMC5883       IOLo(hmc5883CsPin)
 
+static SPI_TypeDef *hmc5883SpiInstance;
 static IO_t hmc5883CsPin = IO_NONE;
 
 bool hmc5883SpiWriteCommand(uint8_t reg, uint8_t data)
```

Comment:
And merge it with I2C version, at least for non-realtime devices (gyro is special case, every tick counts)